tmt/base.py

```diff
@@ -1587,13 +1588,30 @@ def import_plan(self) -> Optional['Plan']:
                 else:
                     tmt.utils.git_clone(plan_id.url, destination, self)
                 if plan_id.ref:
-                    self.run(Command('git', 'checkout', plan_id.ref), cwd=destination)
+                    # attempt to evaluate dynamic reference
+                    try:
+                        ref = resolve_dynamic_ref(
+                            workdir=destination,
+                            ref=plan_id.ref,
+                            plan=self,
+                            logger=self._logger
+                            )
+                    except tmt.utils.FileError as error:
+                        raise tmt.utils.DiscoverError(str(error))
+                    if ref:
+                        # would it hurt if we simply rewrite plan_id.ref?
+                        plan_id._ref_expanded = ref
```

Comment:
No idea about the `plan_id.ref`, I'd probably leave it as it is, although judging just by a feeling.

I think this assignment can be done under `try` already, as `resolve_dynamic_ref` returns `Optional[str]` which is an exact match of `_ref_expanded`, and `plan_id._ref_expanded` can be used instead of `ref`. It seems `ref` itself is not used anywhere else. Or, if you like the extra variable, please consider a slightly different name - it's not obvious there's a difference between `plan_id.ref` and `ref`, so something like `resolved_ref`, or `ref_expanded`, to match the corresponding field of `FmfId`, would help me navigate the code.

Reply:
Updated.